api/agent/agent.go

```diff
@@ -453,12 +482,10 @@ func (s *coldSlot) exec(ctx context.Context, call *call) error {
 	span, ctx := opentracing.StartSpanFromContext(ctx, "agent_cold_exec")
 	defer span.Finish()
 
-	waiter, err := s.cookie.Run(ctx)
-	if err != nil {
-		return err
-	}
```

Comment:
not so sure about moving this into prepare, this is the execution. left comment at other location